packages/fast-css-editor-react/src/spacing/spacing.tsx

```diff
@@ -0,0 +1,222 @@
+                onChange={this.handleInputOnChange(spacingKey)}
+                value={this.props[spacingKey] || ""}
+            />
+        );
+    }
+
+    private getTypeClassNames(spacingType: SpacingType): string {
+        let classes: string = get(this.props, "managedClasses.cssSpacing_type");
+
+        classes += ` ${get(
+            this.props,
+            `managedClasses.cssSpacing_type__${spacingType}`
+        )}`;
+
+        if (spacingType === this.state.activeType) {
+            classes += ` ${get(
+                this.props,
+                `managedClasses.cssSpacing_type__${spacingType}__active`
+            )}`;
+        }
+
+        if (spacingType === this.state.hoverType) {
+            classes += ` ${get(
+                this.props,
+                `managedClasses.cssSpacing_type__${spacingType}__hover`
+            )}`;
+        }
+
+        return classes;
+    }
+
+    private handleTypeClick(
+        spacingType: SpacingType
+    ): (e: React.MouseEvent<HTMLDivElement>) => void {
+        return (e: React.MouseEvent<HTMLDivElement>): void => {
+            if (e.currentTarget === e.target) {
+                if (typeof this.props.onSpacingTypeUpdate === "function") {
+                    this.props.onSpacingTypeUpdate(spacingType);
+                } else if (this.props.spacingType === undefined) {
+                    this.setState({
+                        activeType: spacingType,
+                    });
+                }
+            }
+        };
+    }
+
+    private handleMouseOver(
+        spacingType?: SpacingType
+    ): (e: React.MouseEvent<HTMLDivElement>) => void {
+        return (e: React.MouseEvent<HTMLDivElement>): void => {
+            if (spacingType && e.currentTarget === e.target) {
+                this.setState({
+                    hoverType: spacingType,
+                });
+            } else if (spacingType === undefined) {
+                this.setState({
+                    hoverType: void 0,
+                });
+            }
+        };
+    }
+
+    private handleMouseOut = (e: React.MouseEvent<HTMLDivElement>): void => {
+        if (e.currentTarget === e.target) {
+            this.setState({
+                hoverType: void 0,
+            });
+        }
+    };
+
+    private handleInputOnChange(
+        cssKey: SpacingKey
+    ): (e: React.ChangeEvent<HTMLInputElement>) => void {
+        return (e: React.ChangeEvent<HTMLInputElement>): void => {
+            const spacing: CSSSpacingValues = {};
+
+            Object.keys(this.props).forEach(
+                (key: string): void => {
+                    if (
+                        this.props[key] &&
+                        key !== "spacingType" &&
+                        key !== "managedClasses" &&
+                        typeof this.props[key] !== "function"
+                    ) {
+                        spacing[key] = this.props[key];
+                    }
```

Comment:
what if I put "aria-label" on here? am I going to get that in my update object?

Might be better to use lodash `pick` from the keys that actually satisfy the object structure.

Reply:
Yep, good point